packages/server/lib/controllers/v1/flow/upgrade/putPreBuilt.ts

```diff
@@ -0,0 +1,49 @@
+import { z } from 'zod';
+import { asyncWrapper } from '../../../../utils/asyncWrapper.js';
+import type { UpgradePreBuiltFlow, IncomingFlowConfigUpgrade } from '@nangohq/types';
+import { requireEmptyQuery, zodErrorToHTTP } from '@nangohq/utils';
+import { flowConfig } from '../../../sync/deploy/postConfirmation.js';
+import { logContextGetter } from '@nangohq/logs';
+import { upgradePreBuilt as upgradePrebuiltFlow } from '@nangohq/shared';
+
+const validation = flowConfig.extend({
+    id: z.string(),
+    upgrade_version: z.string(),
+    last_deployed: z.string(),
+    is_public: z.boolean(),
+    pre_built: z.boolean()
+});
+
+export const putUpgradePreBuilt = asyncWrapper<UpgradePreBuiltFlow>(async (req, res) => {
+    const emptyQuery = requireEmptyQuery(req, { withEnv: true });
+    if (emptyQuery) {
+        res.status(400).send({ error: { code: 'invalid_query_params', errors: zodErrorToHTTP(emptyQuery.error) } });
+        return;
+    }
+
+    const val = validation.safeParse(req.body);
+
+    if (!val.success) {
+        res.status(400).send({
+            error: { code: 'invalid_body', errors: zodErrorToHTTP(val.error) }
+        });
+        return;
+    }
+
+    const flowConfig: IncomingFlowConfigUpgrade = val.data as IncomingFlowConfigUpgrade;
```

Comment:
Since zod and types is not synced we tried to assing the body to the type however if you enforce type cast it's defeating the purpose. Here for example `is_public` is not compatible 
You can fix it by using `is_public: z.literal(true),`
```suggestion
    const flowConfig: IncomingFlowConfigUpgrade = val.data;
```

Reply:
Good point, updated